### ISSUE TITLE
Debugging crawl-specs.js is hard due to use of child processes

### ISSUE THREAD

**Issue by foolip** [tidoust/reffy#207]
Getting to the bottom of https://github.com/tidoust/reffy/issues/205 is made more difficult because of the using of child processes in `crawlSpecInChildProcess`. It took me a while to figure out this was even happening.

@tidoust do you have any tricks for debugging? Could a single process mode be supported for easier debugging?

**Comment by tidoust** [comment]
I created a debug mode for Reffy through https://github.com/tidoust/reffy/commit/651057de737b90eea927e24687bec3c490bf6ae4, based on #209. The debug mode also crawls specs sequentially to further ease debugging.

I haven't really found a good way to debug the whole process to be honest. Ironically, I had actually introduced child processes to debug issues: there were weird clashes between JSDOM instances and memory leaks that often made the crawl crash.

I don't think that's still the case so we could perhaps drop the use of child processes altogether. Memory usage may still be an issue though if we crawl specs in parallel within the same process. For instance, if the program needs to crawl two HTML-like specs in parallel, it may need more than 3GB of RAM. Our server was automatically killing processes that used more than 2GB, I haven't checked practical limits with GitHub Actions.

This is somewhat related to #196: if Puppeteer gets used throughout, it would run in dedicated processes by definition, so we wouldn't need to create child processes on our own.
